cli-output/src/cli_output.rs

```diff
@@ -610,25 +610,56 @@ fn show_votes_and_credits(
         return Ok(());
     }
 
-    writeln!(f, "Recent Votes:")?;
-    for vote in votes {
-        writeln!(f, "- slot: {}", vote.slot)?;
-        writeln!(f, "  confirmation count: {}", vote.confirmation_count)?;
-    }
-    writeln!(f, "Epoch Voting History:")?;
     writeln!(
         f,
-        "* missed credits include slots unavailable to vote on due to delinquent leaders",
+        "{} Votes (using {}/{} entries):",
+        (if votes.len() < MAX_LOCKOUT_HISTORY {
+            "All"
+        } else {
+            "Recent"
+        }),
+        votes.len(),
+        MAX_LOCKOUT_HISTORY
     )?;
-    for entry in epoch_voting_history {
+
+    for vote in votes.iter().rev() {
+        writeln!(f, "- slot: {} (confirmation count: {})", vote.slot, vote.confirmation_count)?;
+    }
+    if let Some(newest) = epoch_voting_history.iter().rev().next() {
+        writeln!(
+            f,
+            "- ... (truncated older {} votes, which was rooted and became credits)",
+            newest.credits
+        )?;
+    }
+
+    if !epoch_voting_history.is_empty() {
+        writeln!(
+            f,
+            "{} Epoch Voting History (using {}/{} entries):",
+            (if epoch_voting_history.len() < MAX_EPOCH_CREDITS_HISTORY {
+                "All"
+            } else {
+                "Recent"
+            }),
+            epoch_voting_history.len(),
+            MAX_EPOCH_CREDITS_HISTORY
+        )?;
+        writeln!(
+            f,
+            "* missed credits include slots unavailable to vote on due to delinquent leaders",
+        )?;
+    }
+
+    for entry in epoch_voting_history.iter().rev() {
         writeln!(
             f, // tame fmt so that this will be folded like following
             "- epoch: {}",
             entry.epoch
         )?;
         writeln!(
             f,
-            "  credits range: [{}..{})",
+            "  credits range: ({}..{}]",
```

Comment:
as this `run.sh` sample shows this is more proper range notation... I was wrong at #13374 in hindsight after more deeper understanding of staking:

```
$ ./target/release/solana -ul vote-account 3vZMFHQPt618pUnjEnnw5yXbNgEK6PMzLNHEZrroe8Qn
Account Balance: 0.02685864 SOL
Validator Identity: A9ZHywzYR3G17utyPs8kb5Tm7pcrbuaGC56AWuuWLNEp
Authorized Voters: {0: "A9ZHywzYR3G17utyPs8kb5Tm7pcrbuaGC56AWuuWLNEp"}
Authorized Withdrawer: A9ZHywzYR3G17utyPs8kb5Tm7pcrbuaGC56AWuuWLNEp
Credits: 1
Commission: 100%
Root Slot: 0
Recent Timestamp: 2021-02-10T05:49:58Z from slot 31
Recent Votes (using 31/31 entries):
- slot: 31 (confirmation count: 1)
- slot: 30 (confirmation count: 2)
- slot: 29 (confirmation count: 3)
- slot: 28 (confirmation count: 4)
- slot: 27 (confirmation count: 5)
- slot: 26 (confirmation count: 6)
- slot: 25 (confirmation count: 7)
- slot: 24 (confirmation count: 8)
- slot: 23 (confirmation count: 9)
- slot: 22 (confirmation count: 10)
- slot: 21 (confirmation count: 11)
- slot: 20 (confirmation count: 12)
- slot: 19 (confirmation count: 13)
- slot: 18 (confirmation count: 14)
- slot: 17 (confirmation count: 15)
- slot: 16 (confirmation count: 16)
- slot: 15 (confirmation count: 17)
- slot: 14 (confirmation count: 18)
- slot: 13 (confirmation count: 19)
- slot: 12 (confirmation count: 20)
- slot: 11 (confirmation count: 21)
- slot: 10 (confirmation count: 22)
- slot: 9 (confirmation count: 23)
- slot: 8 (confirmation count: 24)
- slot: 7 (confirmation count: 25)
- slot: 6 (confirmation count: 26)
- slot: 5 (confirmation count: 27)
- slot: 4 (confirmation count: 28)
- slot: 3 (confirmation count: 29)
- slot: 2 (confirmation count: 30)
- slot: 1 (confirmation count: 31)
- ... (truncated older 1 votes, which was rooted and became credits)
All Epoch Voting History (using 1/64 entries):
* missed credits include slots unavailable to vote on due to delinquent leaders
- epoch: 0
  credits range: (0..1]
  credits/slots: 1/128

```